samples/unit_testing/subscription-manager/unit_tests/test_createenvhttpstart_invalidauth.py

```diff
@@ -0,0 +1,75 @@
+import azure.functions as func
+import azure.durable_functions as df
+import unittest.main as unitmain
+from unittest import IsolatedAsyncioTestCase,mock
+from unittest.mock import AsyncMock, MagicMock, patch
+from ..CreateEnvironmentHTTPStart import main
+
+"""
+Test class for CreateEnvironmentHTTPStart Durable HTTP Starter Function that kicks off orchestrations
+Mocks the HTTP Request and expected HTTP Response from the Durable HTTP start method and tests
+- no header authorization
+- header with invalid client principal
+
+Also tests Authorization decorator.
+"""
+class DurableFunctionsHttpStartTestCaseInvalidAuth(IsolatedAsyncioTestCase):
+
+    async def test_durablefunctionsorchestrator_trigger_noheader(self):
+        function_name = 'DurableFunctionsOrchestrator'
+        instance_id = 'f86a9f49-ae1c-4c66-a60e-991c4c764fe5'
+        starter = MagicMock()
+
+        mock_request = func.HttpRequest(
+            method='GET',
+            body=None,
+            url=f'http://localhost:7071/api/orchestrators{function_name}',
+            route_params={'functionName': function_name},
+        )
+
+        mock_response = func.HttpResponse(
+            body = None,
+            status_code= 200,
+            headers={
+                "Retry-After": 10
+            }
+        )
+        
+        with patch('azure.durable_functions.DurableOrchestrationClient',spec=df.DurableOrchestrationClient) as a_mock:
+            a_mock.start_new = AsyncMock()
+            a_mock().start_new.return_value = instance_id
+            a_mock().create_check_status_response.return_value = mock_response
+            response = await main(req=mock_request, starter=starter)
+            self.assertEqual(403,response.status_code)
```

Comment:
Just to make sure I understand, this isn't testing the HTTPTrigger itself, this is actually just trying to test the decorator, right? I'm just confused because it seems to be like the return value for `create_check_status_response` is already being provided

Reply:
Yes we are only testing the decorator here and making sure it authorizes and calls the function. So yes, for valid auth we should be removing the canned response. Will discuss further. For the invalid auth, the decorator itself would return a 403 - so we don't really care about the create_check_status_response. but I see why it can be confusing. Will remove this canned response